q2_taxa/tests/test_visualizer.py

```diff
@@ -43,13 +64,51 @@ def test_barplot(self):
             self.assertTrue(os.path.exists(csv_lvl3_fp))
             self.assertTrue('val1' in open(csv_lvl3_fp).read())
 
+    @patch(
+        'q2_taxa._visualizer._call_relative_frequency',
+        wraps=_call_relative_frequency
+    )
+    def test_barplot_pipeline_absolute_frequency(
+        self, relative_frequency_mock
+    ):
+        table = Artifact.import_data(
+            'FeatureTable[Frequency]', self.table, view_type=biom.Table
+        )
+        self.barplot(table=table)
+
+        # should be normalized
+        relative_frequency_mock.assert_called_once()
+
+    @patch(
+        'q2_taxa._visualizer._call_relative_frequency',
+        wraps=_call_relative_frequency
+    )
+    def test_barplot_pipeline_relative_frequency(
+        self, relative_frequency_mock
+    ):
+        table = Artifact.import_data(
+            'FeatureTable[RelativeFrequency]',
+            self.rel_table,
+            view_type=biom.Table
+        )
+        self.barplot(table=table)
+
+        # should not be redundantly normalized
+        relative_frequency_mock.assert_not_called()
+
     def test_barplot_metadata_extra_id(self):
         metadata = qiime2.Metadata(
             pd.DataFrame({'val1': ['1.0', '2.0', '3.0', '4.0', '5.0']},
                          index=pd.Index(['A', 'B', 'C', 'D', 'E'], name='id')))
 
         with tempfile.TemporaryDirectory() as output_dir:
-            barplot(output_dir, self.table, self.taxonomy, metadata)
+            _barplot(
```

Comment:
I have a bit of a nit-picky comment here that I'm on the fence on even posting, but here goes: I recommend creating the relevant artifacts and testing `barplot` in this (and the other) tests in this section. It makes the most sense to me to use the user-facing function since we wouldn't care, for example, if the validation was moved from `_barplot` to `barplot`, and it's possible that the `_barplot` function wouldn't be needed at all someday. 

@colinvwood, are there motivations that I'm not thinking of here to test the `_barplot` action rather than the `barplot` action?